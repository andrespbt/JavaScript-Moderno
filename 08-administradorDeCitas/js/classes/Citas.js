import UI from './UI.js';
const ui = new UI();
class Citas {
  constructor() {
    this.citas = [];
  }

  // Agregar nueva cita al array

  agregarCita(cita) {
    this.citas = [...this.citas, cita];
  }

  // Eliminar cita del array

  eliminarCitas(id) {
    this.citas = this.citas.filter(cita => cita.id !== id);
  }

  // Crear nuevo array con cita modificada

  editarCita(citaActualizada) {
    this.citas = this.citas.map(cita => (cita.id === citaActualizada.id ? (cita = citaActualizada) : cita));
    ui.agregarCitaHtml(this.citas);
  }
}

export default Citas;
