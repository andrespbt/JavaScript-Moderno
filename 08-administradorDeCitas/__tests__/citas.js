/**
 * @jest-environment jsdom
 */
import Citas from '../js/classes/Citas';

describe('Probar la clase de citas', () => {
  const administrarCitas = new Citas();
  test('Agregar una nueva cita', () => {
    const datosCliente = {
      mascota: 'Hook',
      propietario: 'Andres Poblete',
      telefono: '231313',
      fecha: '10-12-2015',
      hora: '12:20',
      sintomas: 'Solo duerme'
    };

    datosCliente.id = Date.now();

    administrarCitas.agregarCita(datosCliente);

    // Test
    expect(administrarCitas).toMatchSnapshot();
  });
});
