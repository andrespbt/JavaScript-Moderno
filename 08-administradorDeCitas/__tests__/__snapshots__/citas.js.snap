// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2015",
      "hora": "12:20",
      "id": 1663007750878,
      "mascota": "Hook",
      "propietario": "Andres Poblete",
      "sintomas": "Solo duerme",
      "telefono": "231313",
    },
  ],
}
`;
